### PR TITLE
CEO PR#4.1 - CardFinder.ts

### PR DESCRIPTION
**Pull request by d-little**
_PR#1 in a set of card+deck related PRs_

### CardFinder.ts

I pretty much just tried to replicate the code as-is for Preludes, however there are a few oddities I'm not 100% sure on.  

For starters, `getLeaderByName`+`getPreludeByName` are only ever referenced in this `CardFinder.ts` in their respective `leadersFromJSON`+`preludesFromJSON` blocks in this same file.  Not sure why they'd need to be split out?

I've also included `leaderCards` in the generic `getCardByName`, but again, not sure this needed

Prior to this I'd just used the generic `cardsFromJSON` and it worked Fine, but would rather get this right if necessary.

(This whole PR for CardFinder.ts may be unnecessary, but will want to get it ironed out before moving on to `Deck.ts`+`Game.ts`

### CODE REVIEW

File: src/server/CardFinder.ts [hunk 1]
```diff
@@ -75,6 +80,23 @@ export class CardFinder {
     return result;
   }
 
+  public leadersFromJSON(cards: Array<CardName>): Array<ILeaderCard> {
```
Comment (kberg) [inline src/server/CardFinder.ts:83]: This should be reduced to one main function now that it's repeated three times. But it's OK.

**Review by kberg — COMMENTED** [review]
This PR is fine. Add the test. :D

File: src/server/CardFinder.ts (hunk 1 above)
Comment (d-little) [reply src/server/CardFinder.ts:83]: > Add the test. :D

What test ? :o

I can add in some cardfinder tests when there are some cards to find!  But, at the moment, I'm not sure where to add more tests for this new block

**Comment by kberg** [comment]
https://github.com/terraforming-mars/terraforming-mars/blob/restart-turn/tests/CardFinder.spec.ts

**Comment by kberg** [comment]
It's not in the `server` directory for historical reasons.

**Review by kberg — APPROVED** [review]

**Pull request merged**